### ISSUE TITLE
vessel init fails with "error initializing app"

### ISSUE THREAD

**Issue by mtlynch** [Vessel-App/vessel-cli#23]
*Aside: Great project! I'm excited to try it.*

I'm running into an issue where vessel seems to fail initialize a new app.

1. Authenticate successfully with Fly API token (`vessel auth -t [mytoken]`)
2. `vessel init`
3. Choose an app name
4. For image, choose other > `cimg/node:16.14.2`
5. Re-select `cimg/node:16.14.2`

vessel fails with:

```bash
error initializing app
```

I can't find any other diagnostic information, but I'm happy to share if you point me to it. `~/.vessel/debug.log` is empty.

---

**Edit**: I discovered the `--verbose` option, so here's the better error:

```
could not register app: request error: http client error: Post "http://127.0.0.1:4280/v1/apps": dial tcp 127.0.0.1:4280: connect: connection refused
```

It might be worth mentioning that I'm running these commands in a headless VM through SSH.

---

**Edit2**: I was able to work around this by running `fly machines api-proxy` in a separate terminal.

### Environment

```bash
$ vessel --version
0.0.17

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.6 LTS
Release:        18.04
Codename:       bionic

$ uname -a
Linux tp 5.11.22-3-pve #1 SMP PVE 5.11.22-6 (Wed, 28 Jul 2021 10:51:12 +0200) x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by fideloper** [comment]
The code should be starting that proxy for you in the background (and then turning it off when it's done making API calls). It needs to find `flyctl` in $PATH however. 

It also checks to see if the proxy is running already before turning it on - so starting the proxy manually yourself will work as well.

The logic for if it should start the proxy (and how) is here: https://github.com/Vessel-App/vessel-cli/blob/main/internal/fly/api-proxy.go

I'd be curious to see if a different error later stops the command without cleaning up / stopping the proxy.

**Comment by fideloper** [comment]
No worries - I guess I'd try to run `ps aux | grep fly` to see if the proxy is running as like a zombie process or something.

The Go code detecting if it's already running or not (and thus decides if it needs to start the proxy) is running:

```
conn, err := net.Dial("tcp", "127.0.0.1:4280")
	if err != nil {
		// If proxying is not happening unable to connect,
		// we *should* attempt to start the proxy
		return true
	}
```

I wonder if that's not returning an error in your case, and thus tells the code NOT to start the proxy.

**Comment by fideloper** [comment]
(wow that code comment is bad there haha)

**Comment by mtlynch** [comment]
I just tried debugging this on a clean machine and made some progress.

One issue is that `StartMachineProxy` ignores the Fly auth token, so launching the proxy kept failing. The failures were silent due to other errors, which I proposed a fix for in #26.

I don't think that's the root cause, as my earlier test was able to talk to the Fly API successfully because I saw it create a new Machine on Fly.
